utils/post_utils.jsx

```diff
@@ -68,7 +68,22 @@ export function canDeletePost(post) {
     return haveIChannelPermission(store.getState(), {channel: post.channel_id, team: channel && channel.team_id, permission: Permissions.DELETE_OTHERS_POSTS});
 }
 
-export function canEditPost(post, editDisableAction) {
+export function disableCanEditPostByTime(post, editDisableAction) {
+    const canEdit = canEditPost(post);
+    const license = getLicense(store.getState());
+    const config = getConfig(store.getState());
+
+    if (canEdit && license.IsLicensed === 'true') {
+        if (config.PostEditTimeLimit !== '-1' && config.PostEditTimeLimit !== -1) {
+            const timeLeft = (post.create_at + (config.PostEditTimeLimit * 1000)) - Utils.getTimestamp();
+            if (timeLeft > 0) {
+                editDisableAction.fireAfter(timeLeft + 1000);
```

Comment:
I don't like this... it's too magical. Is there a better pattern we can follow here for having a component re-reneder after a certain time?

Reply:
@crspeller You must be referring to the existing code then because the LOC you highlighted isn't new.
@jespino Note that the `UNSET_POST_EDIT_TIME_LIMIT` constant is available.

Reply:
I moved that logic into the component itself, so now is clearer.

Probably move canEditPost and canDeletePost to redux selector would be great.

@mkraft I used it but I compared againts the "string" conversion, because sometimes it is an string, and sometimes it is a number.